File: benchmark/tasks/single-func-fix/verify.mjs
import { pathToFileURL } from "node:url";
import path from "node:path";
import assert from "node:assert/strict";

const workspace = process.argv[2];
if (workspace === undefined) {
    throw new Error("缺少 workspace 参数");
}

const { add } = await import(pathToFileURL(path.resolve(workspace, "src/math.js")).href);

assert.equal(add(2, 3), 5);
assert.equal(add(-1, 1), 0);
assert.equal(add(0, 0), 0);

console.log("single-func-fix 验收通过");

File: benchmark/tasks/single-func-fix/verify.py
import importlib.util
import pathlib
import sys

workspace = pathlib.Path(sys.argv[1])
source = workspace / "src" / "math.py"
spec = importlib.util.spec_from_file_location("math_task", source)
assert spec is not None and spec.loader is not None, "无法加载 math.py"
module = importlib.util.module_from_spec(spec)
sys.modules[spec.name] = module
spec.loader.exec_module(module)

assert module.add(2, 3) == 5
assert module.add(-1, 1) == 0
assert module.add(0, 0) == 0

print("single-func-fix 验收通过")
